pprof/pprof.go

```diff
@@ -0,0 +1,53 @@
+package pprof
+
+import (
+	"context"
+	"github.com/NethermindEth/juno/utils"
+	"net/http"
+	_ "net/http/pprof"
+	"strconv"
+)
+
+type Pprof struct {
+	enabled bool
+	port    uint16
+	log     utils.SimpleLogger
+	server  *http.Server
+}
+
+func New(enabled bool, port uint16, log utils.SimpleLogger) *Pprof {
+	server := &http.Server{
+		Addr:    "localhost:" + strconv.Itoa(int(port)),
+		Handler: http.DefaultServeMux,
+	}
+
+	return &Pprof{
+		enabled: enabled,
+		server:  server,
+		log:     log,
+	}
+}
+
+func (p *Pprof) Run(ctx context.Context) error {
+	if !p.enabled {
+		return nil
+	}
+
+	go func() {
+		p.log.Infow("Starting pprof...", p.server.Addr)
+		if err := p.server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			p.log.Errorw("Pprof server error", "err", err)
+		}
+	}()
+
+	<-ctx.Done()
+
+	p.log.Infow("Shutting down pprof...")
+	err := p.server.Shutdown(context.Background())
```

Comment:
`return p.server.Shutdown(context.Background())` is enough, node will log the error anyways